dbms/src/Common/getNumberOfPhysicalCPUCores.cpp

```diff
@@ -15,6 +17,22 @@ unsigned getNumberOfPhysicalCPUCores()
 {
 #if defined(__x86_64__)
 
+    std::ifstream cgroup_read_in("/sys/fs/cgroup/cpu/cpu.cfs_quota_us");
+    if (cgroup_read_in.is_open())
+    {
+        std::string allocated_cpus_share_str{ std::istreambuf_iterator<char>(cgroup_read_in), std::istreambuf_iterator<char>() };
+        int allocated_cpus_share_int = std::stoi(allocated_cpus_share_str);
+
+        cgroup_read_in.close();
+
+        // If a valid value is present
+        if(allocated_cpus_share_int > -1)
```

Comment:
`> -1` looks less clear than `>= 0`

Reply:
The default value if the limit is not set is -1, hence used the value. However, using > 0 makes sense since that will allow us to avoid corner cases where the CPU limit is read as 0